atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterIntegrationTest.java

```diff
@@ -0,0 +1,374 @@
+        });
+    }
+
+    private Set<TableReference> listAllTablesWithMetadata() {
+        return runWithConnection(connection -> {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT table_name from " + AtlasDbConstants.DEFAULT_ORACLE_METADATA_TABLE);
+            ResultSet resultSet = statement.executeQuery();
+            return getTableNamesFromResultSet(resultSet).stream()
+                    .map(TableReference::createUnsafe)
+                    .collect(Collectors.toSet());
+        });
+    }
+
+    private Set<String> listAllTablesWithMapping() {
+        return runWithConnection(connection -> {
+            PreparedStatement statement =
+                    connection.prepareStatement("SELECT table_name from " + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE);
+            ResultSet resultSet = statement.executeQuery();
+            return getTableNamesFromResultSet(resultSet);
+        });
+    }
+
+    private <T> T runWithConnection(FunctionCheckedException<Connection, T, SQLException> task) {
+        try (Connection connection = connectionManager.getConnection()) {
+            return task.apply(connection);
+        } catch (SQLException e) {
+            throw Throwables.rewrapAndThrowUncheckedException(e);
+        }
+    }
+
+    private static Set<String> getTableNamesFromResultSet(ResultSet resultSet) throws SQLException {
+        ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+        while (resultSet.next()) {
+            builder.add(resultSet.getString("table_name"));
+        }
+        return builder.build();
+    }
+
+    private static String withEscapedUnderscores(String s) {
+        return s.replace("_", "\\_");
+    }
+
+    private static String withWildcardSuffix(String s) {
+        return s + "%";
+    }
+
+    private static void assertSnapshotContainsNewTableReferences(
+            AllTableDetailsSnapshot snapshot, int expectedNumberOfNewTableReferences) {
+        assertThat(snapshot.prefixedTableNames()).hasSize(expectedNumberOfNewTableReferences * 2);
+        assertThat(snapshot.physicalTableNames()).hasSize(expectedNumberOfNewTableReferences * 2);
+        assertThat(snapshot.tableReferences()).hasSize(expectedNumberOfNewTableReferences);
+    }
+
+    private static void assertSnapshotIsEmpty(AllTableDetailsSnapshot snapshot) {
+        assertThat(snapshot.prefixedTableNames())
+                .hasSameSizeAs(snapshot.physicalTableNames())
+                .hasSameSizeAs(snapshot.tableReferences())
+                .isEmpty();
+    }
+
+    private static void assertSnapshotIsNotEmpty(AllTableDetailsSnapshot snapshot) {
+        assertThat(snapshot.prefixedTableNames()).isNotEmpty();
+        assertThat(snapshot.physicalTableNames()).isNotEmpty();
+        assertThat(snapshot.tableReferences()).isNotEmpty();
+    }
+
+    @Value.Immutable
+    interface AllTableDetailsSnapshot {
```

Comment:
Helper immutable to make testing that we're cleaning up _everything_ required easier! kinda related to the TableDetails in sqliteoracletable which represents info for a single table, but the issue here is we can't trivially construct a mapping each prefixed table to a physical table to a table reference (without pulling in the table mappers and stuff), but nor do I really need that granularity for these tests.